starfish/src/main/java/org/maxandersen/Starfish.java

```diff
@@ -0,0 +1,238 @@
+            System.out.println("\nThis is the first time you're using starfish...You'll need to  configure it");
+            editConfig(); //Calling function that lets user to configure 
+        }
+        
+        System.out.println("\nLoading configurations.....\n");
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    
+        Config config = mapper.readValue(configFile, Config.class);
+        
+        ide=config.ide;
+        clone_path=config.clone_path;
+        if(ide==null||clone_path==null)editConfig(); //Incase of absence of configuration in file Launch Config 
+        
+        
+    }
+    catch (Exception e) {
+        e.printStackTrace();
+        System.out.print("Press any key to continue . . . ");
+    }
+    
+
+
+    //Clonnning Git Repo
+    //Expected parameter: https://github.com/user-name/repo-name.git
+    String repo_name=args[0].substring(args[0].lastIndexOf("/"),args[0].lastIndexOf(".")); //Extracts the Name of Repository
+    //String clone_path= "/home/fahad/MyProjects/starfish_clonned/"; 
+    String cmd = "git clone "+args[0];
+    String originUrl = args[0];
+    Path directory = Paths.get(clone_path+repo_name);
+    gitClone(directory, originUrl);
+    
+
+
+
+    //Launching Vscode on the Cloned Directory 
+    System.out.println("Launching  IDE Now");
+    runCommand(directory.getParent(), ide,clone_path+repo_name);
+    return 10;
+    
+}
+
+
+//Function to edit configuration and serves for command line starfish config
+public static void editConfig()throws Exception{
+System.out.println("\n------Starfish Configuration Editor------");
+String clone_path="";//Holds path to  destination Where the Repository Must Be Clonned
+    String ide="";  //Holds command for IDE to open upon
+    String userHome = System.getProperty( "user.home" ); //Get User Home Directory: /home/user_name
+
+    File configFile = new File(userHome + "/starfish-config.yml"); //Loading YAML
+    
+    BufferedReader reader= new BufferedReader(new InputStreamReader(System.in));
+    
+    
+    //Reading Configuration File
+    try {
+        if (!configFile.exists()) {
+            //Incase no config file exists
+            System.out.println("\n-No configurations exist for starfish...You'll have to  configure it");
+            System.out.println("\n-New  configuration file will be created at: " + configFile.getAbsolutePath() + "\n");
+            configFile.createNewFile();
+            }
+            
+            
+            //--------------------------First we'll input preferred IDE from user------------------------------
+
+            int id=0;
+            while(id!=1||id!=2||id!=3){
+            System.out.println("\n--------Chose the preferred IDE --------\n 1.for vscode \n 2.for eclipse \n 3.for IntelliJ_IDEA ");
+            
+            id=Integer.parseInt(reader.readLine());
+            
+            if(id==1){ide="code";System.out.println("Selected IDE:VsCode");break;}
```

Comment:
Are these actual ides of editor executables ?

Reply:
Yes